newton/sim/builder.py

```diff
@@ -3591,8 +3671,85 @@ def finalize(self, device: Devicelike | None = None, requires_grad: bool = False
             m.up_axis = self.up_axis
             m.up_vector = np.array(self.up_vector, dtype=wp.float32)
 
+            m.contact_reporter = None
+            if self.contact_queries:
+                self._build_contact_sensors(m)
+
             return m
 
+    def _create_sensor(
```

Comment:
Do we need this separate method? It also seems very specific to `add_contact_sensor`, can't we move this code there?